components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -420,6 +421,90 @@ async fn download_package(req: HttpRequest,
                        &conn)
     {
         Ok(package) => {
+            let channels = match channels_for_package_ident(&req, &package.ident, target, &conn) {
+                Ok(channels) => channels,
+                Err(err) => {
+                    return HttpResponse::InternalServerError()
+                        .body(format!("Failed to determine package channels: {}", err));
+                }
+            };
+
+            let should_restrict = if let Some(chs) = channels.as_ref() {
+                let chs_lower: Vec<String> = chs.iter().map(|c| c.to_lowercase()).collect();
```

Comment:
I'm pretty sure channel names are case sensitive so we should not compare with case insensitivity.